vpr/src/util/vpr_utils.h

```diff
@@ -155,7 +155,23 @@ void print_usage_by_wire_length();
 
 AtomBlockId find_memory_sibling(const t_pb* pb);
 
+/**
+ * @brief Syncs the logical block pins corresponding to the input iblk with the corresponding chosen physical tile
+ * @param iblk cluster block ID to sync within the assigned physical tile
+ *
+ * This routine updates the physical pins vector of the place context after the placement step
+ * to syncronize the pins related to the logical block with the actual connection interface of
+ * the belonging physical tile with the RR graph.
+ *
+ * This step is required as the logical block can be placed at any compatible sub tile locations
+ * within a physical tile.
+ * Given that it is possible to have equivalent logical blocks within a specific sub tile, with
+ * a different subset of IO pins, the various pins offsets must be correctly computed and assigned
+ * to the physical pins vector, so that, when the net RR terminals are computed, the correct physical
+ * tile IO pins are selected.
+ */
```

Comment:
Does it get the location of block iblk from the netlist (I assume it does)?  I assume it needs the block to be placed at a legal (x,y,isub) but it is OK if there is some overuse of the grid (i.e. it does not need the grid_blocks data to be up to date) -- is that right?
This is relevant for things like analytic placers which produce illegal placements in multiple ways (block.x and .y at a location of the wrong type, overused grid_blocks).  I am guessing the block.x, block.y at the wrong type of tile/subtile would be fatal, but it is OK for grid_blocks to not be up to date.

Reply:
So, this routine does not check for illegal placement, in the sense that it just looks for the block and, given it's placement location derives the value of the physical pins to make it match with the pins of the logical block.

It does not strictly check whether the block is compatible with the sub tile and it does not require that a physical location gets overused.

If I remember correctly, all the checks are done prior to this step, so this routine assumes that placement is legal.

Reply:
It's worth writing this in; the routine uses the .x,.y and .sub_tile of the clb netlist, and expects those to put each netlist block at a legal location that can accomodate it. It does not check for overuse of locations, so it can be used with placements that have resource overuse (e.g. global placement algorithms).

Reply:
Sure, comment added.